packages/data-context/src/codegen/templates/react-component/react-component.ejs

```diff
@@ -0,0 +1,13 @@
+---
+fileName: <%= fileName %>
+---
+
+import React from 'react'
+import { <%- componentName %> } from '<%- componentPath %>'
```

Comment:
What happens for default export components, like `CounterDefault`? Would this become `import { CounterDefault } from './counter-default'` (which would be undefined). Did we find a solution for this, or are we moving forward with this as a known limitation?

It should be possible to write a quick babel parser to find the default export and compare that to the components `react-docgen` found. Not sure how expensive this would be, probably not that much.

Reply:
I think we'll need a separate spec template for default export components. I'm looking into how we can figure out if a component is exported by using a custom [handler](https://github.com/reactjs/react-docgen#handlers-1) with React Docgen.

Otherwise yeah we can probably do it after we parse with React Docgen

Reply:
I looked at the result the `componentDefinition` which is what the handler receives and it doesn't contain enough info to determine how the component was exported. For example:
```tsx
const Button = (props: {text: string}) => <button>{props.text}</button>

export { Button }
export default Button
```

The componentDef only contains info on `const Button = (props: {text: string}) => <button>{props.text}</button>`. Rather than a custom handler, I believe a custom resolver makes sense here so we can scrape all of the exported declarations and then compare them to the returned componentDefinitions. We don't need to write our own parser since a resolver directly receives the parsed ast.

Reply:
Going to handle this as a separate task (to be completed before releasing this feature) https://github.com/cypress-io/cypress/issues/24929